### PR TITLE
move InstallationTests to Tests folder

### PR DESCRIPTION
**Pull request by aboedo**
Yet another step in cleaning up the root folder structure for the repo. 
This follows up #1396 and #1402 by moving all InstallationTests to the Tests folder.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -331,25 +331,25 @@ jobs:
...
+          directory: Tests/InstallationTests/CocoapodsInstallation
       - run:
           name: Install Pods
-          working_directory: InstallationTests/CocoapodsInstallation/
+          working_directory: Tests/InstallationTests/CocoapodsInstallation/
           command: |
               pod install
       - scan-and-archive:
-          directory: InstallationTests/CocoapodsInstallation
+          directory: Tests/InstallationTests/CocoapodsInstallation
```
Comment (joshdholtz) [inline .circleci/config.yml:341]: @vegaro This is going to require some fixing when monorepo happens. I've create a story for this in the monorepo epic - https://revenuecats.atlassian.net/browse/CF-446

This is totally fine but just wanted to point this out 😊

File: Tests/InstallationTests/CocoapodsInstallation/fastlane/Fastfile [hunk 2]
```diff
@@ -0,0 +1 @@
+import("../../../../fastlane/Fastfile")
```
Comment (joshdholtz) [inline Tests/InstallationTests/CocoapodsInstallation/fastlane/Fastfile:1]: Do these installation tests need something from the main `Fastfile`? 🤔

File: Tests/InstallationTests/XcodeDirectInstallation/fastlane/Fastfile [hunk 3]
```diff
@@ -0,0 +1 @@
+import("../../../../fastlane/Fastfile")
```
Comment (joshdholtz) [inline Tests/InstallationTests/XcodeDirectInstallation/fastlane/Fastfile:1]: Same question here

**Review by joshdholtz — APPROVED** [review]
Just one small question otherwise this looks good!

File: .circleci/config.yml (hunk 1 above)
Comment (aboedo) [reply .circleci/config.yml:341]: oooff didn't realize I'd break that

File: Tests/InstallationTests/CocoapodsInstallation/fastlane/Fastfile (hunk 2 above)
Comment (aboedo) [reply Tests/InstallationTests/CocoapodsInstallation/fastlane/Fastfile:1]: 🤔 I'm not sure they do anymore!! They did at one point re-use some of the logic from the main Fastfile. 

I'm going to try removing this bit in a separate PR

**Pull request merged**